Modules/Sources/BuildSettingsKit/BuildSettingsEnvironment.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+enum BuildSettingsEnvironment {
+    case live
+    case preview
+
+    static let current: BuildSettingsEnvironment = {
+#if DEBUG
+        let environment = ProcessInfo.processInfo.environment
+        if environment["XCODE_RUNNING_FOR_PREVIEWS"] == "1" {
+            return .preview
+        }
+        if NSClassFromString("XCTestCase") != nil {
+            fatalError("BuildSettings are unavailable when running unit tests. Make sure to inject the values manually in system under test.")
+        }
```

Comment:
What about Swift Testing tests?

Reply:
Good point. I initially ignored it, but now updated based on the code from from https://github.com/pointfreeco/swift-issue-reporting/blob/main/Sources/IssueReporting/IsTesting.swift.